src/Reflection/ReflectionClass.php

```diff
@@ -1014,10 +1016,22 @@ function (Node\Name $interfaceName) {
             : [];
     }
 
+
+    private function getInheritanceClassHierarchy()
+    {
+        if ($this->hierarchy) {
+            return $this->hierarchy;
```

Comment:
@dantleech yes, that's what I suggested to @asgrim to avoid having a mess of complexity. Not sure how well it plays with the entire reflection traversal...